x/lscosmos/keeper/delegation_strategy.go

```diff
@@ -26,6 +26,10 @@ func (k Keeper) DelegateMsgs(ctx sdk.Context, delegatorAddr string, amount sdk.I
 		return nil, err
 	}
 
+	// sort the val address amount based on address to avoid generating different lists
+	// by all validators
+	sort.Sort(valAddressAmount)
```

Comment:
would rather have FetchValidators to delegate return determinsitic ordering.
That is the real black box.
DelegateMsgs and UndelegateMsgs is an abstraction on top.

Reply:
also possibly have `updateValList, updatedDelegationState := k.GetAllValidatorsState(ctx)` also have deterministic ordering

Reply:
We can sort `k.GetAllValidatorsState`  based on the string